gengo/gengo.py

```diff
@@ -415,16 +415,27 @@ def _raiseForMultipleErrorResponse(self, results):
         error_code = error_codes[0] if error_codes else None
         raise GengoError(' '.join(messages), error_code)
 
-    def _raiseForSingleErrorResponse(self, results):
-        raise GengoError(results['err']['msg'], results['err']['code'])
+    def _raiseForSingleErrorResponse(self, results, code):
 
-    def _raiseForErrorResponse(self, results):
+        code = self._checkForCodeInPayload(results, code)
+        raise GengoError(results['err']['msg'], code)
+
+    def _checkForCodeInPayload(self, results, status_code):
+        try:
+            return results['err']['code']
+        except KeyError:
+            return status_code
+
+    def _raiseForErrorResponse(self, results, status_code):
         # See if we got any errors back that we can cleanly raise on
         if 'opstat' in results and results['opstat'] != 'ok':
             # In cases of multiple errors, the keys for results['err'] will be
             # the job IDs.
-            if 'msg' in results['err']:
-                self._raiseForSingleErrorResponse(results)
+            msg = "Internal Server Error"
+            if 'err' not in results:
+                raise GengoError(msg, status_code)
+            elif 'msg' in results['err']:
+                self._raiseForSingleErrorResponse(results, status_code)
             else:
                 self._raiseForMultipleErrorResponse(results)
```

Comment:
we can even remove `else:` so `self._raiseForMultipleErrorResponse(results)` would be the default.